pkg/storageos/update_status.go

```diff
@@ -57,9 +57,54 @@ func (s *Deployment) getStorageOSStatus() (*storageosv1.StorageOSClusterStatus,
 	// Everything is empty if join token is empty.
 	if len(s.stos.Spec.Join) > 0 {
 		nodeIPs = strings.Split(s.stos.Spec.Join, ",")
-		totalNodes = len(nodeIPs)
 	}
 
+	if s.nodev2 {
+		return s.getStorageOSV2Status(nodeIPs)
+	}
+	return s.getStorageOSV1Status(nodeIPs)
+}
+
+// getStorageOSV2Status returns the current v2 cluster's health by probing each
+// node provided.
+//
+// Note: NodeHealthStatus is deprecated and not set for v2.
+func (s *Deployment) getStorageOSV2Status(nodeIPs []string) (*storageosv1.StorageOSClusterStatus, error) {
+	var readyNodes int
+
+	totalNodes := len(nodeIPs)
+	memberStatus := new(storageosv1.MembersStatus)
+
+	for _, ip := range nodeIPs {
+		if isListening(ip, storageosapi.DefaultPort, 2*time.Second) {
```

Comment:
Let's use a constant time for the timeout passed here and in the v1 `getNodeHealth()`.

Reply:
yes, done